nomad/structs/csi.go

```diff
@@ -729,7 +731,9 @@ func (p *CSIPlugin) AddPlugin(nodeID string, info *CSIInfo) error {
 				p.NodesHealthy -= 1
 			}
 		}
-		p.Nodes[nodeID] = info
+		if prev != nil || prev == nil && info.Healthy {
```

Comment:
nitpick: isn't this the same as `prev != nil || info.Healthy`

Reply:
Yes! Will fix.